app/Controllers/Http/AuthController.ts

```diff
@@ -26,12 +40,100 @@ export default class AuthController {
     return { user, token }
   }
 
+  /*
+  |--------------------------------------------------------------------------
+  | Login
+  |--------------------------------------------------------------------------
+  |
+  | This method is in charge of handling login requests.
+  |
+  */
   public async login({ auth, request }: HttpContextContract) {
-    const email = request.input('email')
-    const password = request.input('password')
+    const { email, password, grantDeveloperAccess } = await request.validate({
+      schema: schema.create({
+        email: schema.string({ trim: true }, [rules.email()]),
+        password: schema.string({ trim: true }),
+        grantDeveloperAccess: schema.boolean.optional(),
+      }),
+    })
 
     const token = await auth.use('api').attempt(email, password)
+    // When logging in to the dev portal, grantDeveloperAccess adds the developer role to the user.
+    // Used for analytics and regulation only.
+    const user = await User.findBy('email', email)
+    if (grantDeveloperAccess && !user!.roles.includes('developer')) {
+      user!.roles.push('developer')
+      await user?.save()
+    }
 
     return token
   }
+
+  /*
+  |--------------------------------------------------------------------------
+  | Request password reset
+  |--------------------------------------------------------------------------
+  |
+  | This method is in charge of handling password reset requests.
+  | It will send an email to the user with a reset password link.
+  |
+  */
+  public async requestPasswordReset({ request }: HttpContextContract) {
+    const { email } = await request.validate({
+      schema: schema.create({
+        email: schema.string({ trim: true }, [rules.email()]),
+      }),
+    })
+
+    const user = await User.findByOrFail('email', email)
+
+    const token = string.generateRandom(64)
+    user.resetPasswordToken = token
+    await user.save()
+
+    // TODO: send email to user with a link like this:
+    // https://app.op-ent.fr/auth/reset-password?token=${token}
+  }
+
+  /*
+  |--------------------------------------------------------------------------
+  | Is reset password token valid
+  |--------------------------------------------------------------------------
+  |
+  | This method is in charge of checking if the reset password token
+  | exists.
+  |
+  */
+  public async isResetPasswordTokenValid({ request }: HttpContextContract) {
+    const { token } = await request.validate({
+      schema: schema.create({
+        token: schema.string([rules.length(64)]),
+      }),
+    })
+
+    const user = await User.findByOrFail('resetPasswordToken', token)
```

Comment:
```suggestion
    const user = await User.findBy('resetPasswordToken', token)
```

Reply:
If the token isn't valid, a boolean should be returned instead of an error message.

Reply:
Oh yeah I missed this one thx 👍